File: gameserver/src/main/java/l2s/gameserver/skills/effects/p_passive.java
package l2s.gameserver.skills.effects;

import l2s.gameserver.model.Creature;
import l2s.gameserver.model.Skill;
import l2s.gameserver.model.actor.instances.creature.Abnormal;
import l2s.gameserver.model.instances.NpcInstance;
import l2s.gameserver.templates.skill.EffectTemplate;

public class p_passive extends Abnormal
{
	public p_passive(Creature creature, Creature target, Skill skill, boolean reflected, EffectTemplate template)
	{
		super(creature, target, skill, reflected, template);
	}

	@Override
	public void onStart()
	{
		super.onStart();
		if(getEffected().isNpc())
		{
			NpcInstance npc = (NpcInstance) getEffected();
			npc.setUnAggred(true);
		}
	}

	@Override
	public void onExit()
	{
		super.onExit();
		if(getEffected().isNpc())
			((NpcInstance) getEffected()).setUnAggred(false);
	}
}

File: gameserver/src/main/java/l2s/gameserver/handler/effects/impl/pump/retail/p_passive.kt
package l2s.gameserver.handler.effects.impl.pump.retail

import l2s.gameserver.handler.effects.EffectHandler
import l2s.gameserver.model.Creature
import l2s.gameserver.model.actor.instances.creature.Abnormal
import l2s.gameserver.templates.skill.EffectTemplate

/**
 * Passive effect implementation.
 * TODO should work like this ?
 * @author Adry_85
 * @author Java-man
 */
class p_passive(template: EffectTemplate) : EffectHandler(template) {

    override fun checkPumpCondition(abnormal: Abnormal?, caster: Creature, target: Creature): Boolean {
        return target.isMonster
    }

    override fun pumpStart(abnormal: Abnormal?, caster: Creature, target: Creature) {
        target.stopActions()
        target.flags.disableAllSkills()
        target.flags.immobilized.start()
    }

    override fun pumpEnd(abnormal: Abnormal?, caster: Creature, target: Creature) {
        target.flags.enableAllSkills()
        target.flags.immobilized.stop()
    }

}